ddtrace/opentelemetry/tracer_test.go

```diff
@@ -85,6 +89,74 @@ func TestTracerOptions(t *testing.T) {
 	assert.Contains(fmt.Sprint(sp), "dd.env=wrapper_env")
 }
 
+func TestSpanContext(t *testing.T) {
+	assert := assert.New(t)
+	tp := NewTracerProvider()
+	defer tp.Shutdown()
+	otel.SetTracerProvider(tp)
+	tr := otel.Tracer("")
+
+	pCtx, parent := tr.Start(context.Background(), "parent")
+	pSpanCtx := parent.SpanContext()
+
+	_, child := tr.Start(pCtx, "child")
+	cSpanCtx := child.SpanContext()
+
+	assert.Equal(cSpanCtx.TraceFlags(), pSpanCtx.TraceFlags())
+	assert.Equal(cSpanCtx.TraceID(), pSpanCtx.TraceID())
+	assert.Equal(cSpanCtx.IsRemote(), pSpanCtx.IsRemote())
+	assert.Equal(cSpanCtx.TraceState().String(), pSpanCtx.TraceState().String())
+}
+
+func TestForceFlush(t *testing.T) {
+	assert := assert.New(t)
+	testData := []struct {
+		timeOut time.Duration
+		flushed bool
+	}{
+		{timeOut: 0 * time.Second, flushed: false},
+		{timeOut: 300 * time.Second, flushed: true},
+	}
+	success := false
+	setFlushFail := func(ok bool) { success = ok }
+
+	for _, tc := range testData {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		tp, payloads, cleanup := mockTracerProvider(t)
+		tr := otel.Tracer("")
+		_, sp := tr.Start(context.Background(), "test_span")
+		sp.End()
+		tp.ForceFlush(tc.timeOut, setFlushFail)
+		payload, err := waitForPayload(ctx, payloads)
+		if tc.flushed {
+			if err != nil {
+				t.Log(err)
+				t.Fail()
+			}
+			assert.Contains(payload, "test_span")
+		}
+		assert.Equal(tc.flushed, success)
+		success = false
```

Comment:
Can this just be a part of the test case, instead of being toggled in the test like this? i.e. another field of the `testData` struct

Reply:
I think the default value of `success` will be the same regardless of the test case, but I agree toggling it at the end of each test case is confusing. What do you think of changing the `success` variable and `setFlushFail` function to something like:
```
	UNSET := 0
	ERROR := 1
	OK := 2
	for _, tc := range testData {
		t.Run( ...
			flushStatus := UNSET
			setFlushStatus := func(ok bool) {
				if ok {
					flushStatus = OK
				} else {
					flushStatus = ERROR
				}
			}

```
And then checking:
```
			assert.NotEqual(UNSET, flushStatus)
			if tc.flushed {
				assert.Equal(OK, flushStatus)
			} else {
				assert.Equal(ERROR, flushStatus)
			}
```
at the end?